### PR TITLE
Add release workflows

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

**Review by trask — APPROVED** [review]

File: .github/workflows/patch-release-build.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+# release tag.
+name: Patch Release Build
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: The version to tag the release with, e.g., 1.2.1, 1.2.2
+        required: true
+      commits:
+        description: Comma separated list of commit shas to cherrypick
```
Comment (mateuszrzeszutek) [inline .github/workflows/patch-release-build.yml:11]: We're not doing any cherrypicks in this PR yet, is that correct?

File: .github/workflows/release-build.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        name: Integration Tests
+        with:
+          arguments: --stacktrace integrationTest
+          remote-build-cache-proxy-enabled: false
+      - uses: actions/upload-artifact@v2
+        name: Save integrationTest results
+        if: always()
+        with:
+          name: integration-test-results
+          path: jmx-metrics/build/reports/tests/test
```
Comment (mateuszrzeszutek) [inline .github/workflows/release-build.yml:56]: Just checking: is the report path for `integrationTest` the same as for `test`?

File: .github/workflows/release-build.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        name: Publish
+        with:
+          arguments: --stacktrace final closeAndReleaseSonatypeStagingRepository -Prelease.version=${{ github.event.inputs.version }}
+        env:
+          SONATYPE_USER: ${{ secrets.SONATYPE_USER }}
+          SONATYPE_KEY: ${{ secrets.SONATYPE_KEY }}
+          GRGIT_USER: ${{ github.actor }}
+          GRGIT_PASS: ${{ secrets.GITHUB_TOKEN }}
+          GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
+          GPG_PASSWORD: ${{ secrets.GPG_PASSWORD }}
```
Comment (mateuszrzeszutek) [inline .github/workflows/release-build.yml:80]: Nit: should we create a GH release here?

File: .github/workflows/patch-release-build.yml (hunk 1 above)
Comment (anuraaga) [reply .github/workflows/patch-release-build.yml:11]: Nah we're not doing any releasing in this PR, just adding workflows

File: .github/workflows/release-build.yml (hunk 2 above)
Comment (anuraaga) [reply .github/workflows/release-build.yml:56]: Right we currently override the default `test` task. I want to change to test-sets eventually

File: .github/workflows/release-build.yml (hunk 3 above)
Comment (anuraaga) [reply .github/workflows/release-build.yml:80]: Oh yeah, added

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**